plugins/woocommerce/client/blocks/tests/e2e/utils/editor/editor-utils.page.ts

```diff
@@ -105,8 +105,19 @@ export class Editor extends CoreEditor {
 		await this.page
 			.getByRole( 'menuitem', { name: /Reset|Delete/ } )
 			.click();
+
 		await this.page.getByRole( 'button', { name: /Reset|Delete/ } ).click();
 
+		await this.page.waitForResponse(
+			( response ) =>
+				( response.url().includes( 'wp-json/wp/v2/templates' ) ||
+					response
+						.url()
+						.includes( 'wp-json/wp/v2/template-parts' ) ) &&
+				response.status() === 200 &&
+				response.request().method() === 'POST'
+		);
```

Comment:
There is still a chance when the request on click is completed before `waitForResponse` start listening to it. Start listening before clicking would be more reliable:
```suggestion
		const responsePromise = this.page.waitForResponse(
			( response ) =>
				( response.url().includes( 'wp-json/wp/v2/templates' ) ||
					response
						.url()
						.includes( 'wp-json/wp/v2/template-parts' ) ) &&
				response.status() === 200 &&
				response.request().method() === 'POST'
		);

		await this.page.getByRole( 'button', { name: /Reset|Delete/ } ).click();

		await responsePromise;
```

Reply:
Good point! Thanks! 🙇

Reply:
Fixed with d5e0657129e997e21142a789efe09f9290388df5